tests/test_architecture.py

```diff
@@ -11,6 +11,22 @@
 GNAME = "framework-testing"
 
 
+def summary_exception(values: np.ndarray,
+                      raise_exception: bool = False) -> int:
+    if raise_exception:
+        raise ValueError("Summary exception raised.")
+
+    return len(values)
+
+
+def summary_memory_error(values: np.ndarray,
+                         raise_mem_err: bool = False) -> int:
+    if raise_mem_err:
+        aux = np.zeros(int(1e+20), dtype=np.float64)
```

Comment:
What do you think of creating a function called 'memory error' to return this memory error statement? I think it will make the code more understandable.

Reply:
I agree. Should I implement it as a class method or a external function?

Reply:
I think an external function, however, to avoid code pollution, it is better to put it in util.py

What do you think?

Reply:
Nice. I'll work on this.